src/jpgimage.cpp

```diff
@@ -668,23 +669,23 @@ namespace Exiv2 {
                             // and dumping the XMP in a post read operation similar to kpsIptcErase
                             // for the moment, dumping 'on the fly' is working fine
                             if (!bExtXMP) {
-                                while (xmp[start])
+                                while (xmp.at(start))
```

Comment:
[question] Did you use here the method `at()` instead of the `operator[]` intentionally? Since we now the size of the `xmp` buffer we could also check in the loop if the start index increase over the size of the buffer.

Reply:
Yes, because if `xmp` does not contain a `0`, the loop will eventually read beyond the bounds of `xmp`.

Reply:
It's fine as you did it. I guess the performance penalty of using `at` instead of  `operator[]` is negligible here.

Reply:
The significantly larger performance penalty will unfortunately be the call to the constructor of `std::string`.